src/Plutus/Types/Transaction.purs

```diff
@@ -45,13 +49,10 @@ instance ToData TransactionOutput where
   toData (TransactionOutput { address, amount, dataHash }) =
     Constr zero [ toData address, toData amount, toData dataHash ]
 
-newtype UtxoM = UtxoM Utxo
+type UtxoMap = Map TransactionInput TransactionOutput
 
-derive instance Generic UtxoM _
-derive instance Newtype UtxoM _
-derive newtype instance Eq UtxoM
-
-instance Show UtxoM where
-  show = genericShow
-
-type Utxo = Map TransactionInput TransactionOutput
+lookupTxHash
+  :: TransactionHash -> UtxoMap -> Array (TransactionInput /\ TransactionOutput)
+lookupTxHash txHash utxos =
+  filter (fst >>> unwrap >>> _.transactionId >>> eq txHash) $
+    Map.toUnfoldable utxos
```

Comment:
We can consider to return an array of `Plutus.Types.TransactionUnspentOutput`s there and also provide accessors (lenses?) for `TransactionUnspentOutput`, so that, for example, retrieving the input of the first uspent output can easily be expressed using a combination of lenses.